Objects/floatobject.c

```diff
@@ -246,6 +246,15 @@ PyFloat_AsDouble(PyObject *op)
 
     nb = Py_TYPE(op)->tp_as_number;
     if (nb == NULL || nb->nb_float == NULL) {
+        if (nb && nb->nb_index) {
+            PyObject *res = PyNumber_Index(op);
+            if (!res) {
+                return -1;
```

Comment:
Super-nitpick: I'd prefer `return -1.0;` (leaving the return-type conversion implicit feels odd to me), but I see that we're returning plain-old `1` in the other branch, so I guess this is fine.